packages/components/button/src/types.ts

```diff
@@ -14,3 +16,69 @@ export type ButtonStylesProps = {
   isDisabled: boolean;
   isFullWidth: boolean;
 };
+
+interface BaseButtonProps extends CommonProps {
+  children?: React.ReactNode;
+  /**
+   * Determines style variation of Button component
+   * @default secondary
+   */
+  variant?: ButtonVariant;
+  /**
+   * Determines size variation of Button component
+   * @default medium
+   */
+  size?: ButtonSize;
+  /**
+   * Applies active styles
+   * @default false
+   */
+  isActive?: boolean;
+  /**
+   * Disabled interaction and applies disabled styles
+   * @default false
+   */
+  isDisabled?: boolean;
+  /**
+   * Adds dropdown indicator icon
+   */
+  isDropdown?: boolean;
+  /**
+   * Expects any of the icon components
+   */
+  icon?: React.ReactElement;
+  /**
+   * Adds loading indicator icon and disables interactions
+   */
+  isLoading?: boolean;
+  /**
+   * Forces button to take 100% of the container
+   */
+  isFullWidth?: boolean;
+  /**
+   * The element used for the root node.
+   * @default button
+   */
+  as?: 'a' | 'button';
+  /**
+   * Button html type attribute
+   * @default button
+   */
+  type?: 'submit' | 'button' | 'reset';
```

Comment:
I guess that both `type` and `href` should be automatically interited from `as` property:

- `type` if `as="button"`
- `href`, `rel`, etc if as="a"

Reply:
Not sure if you need to specify it here explicitly